.github/workflows/test-with-pip.yml

```diff
@@ -0,0 +1,77 @@
+# This workflow targets stable released dependencies from PyPI
+# The matrix is conservative to avoid using too much public resources.
+
+name: Test with PyPI
+
+on: push
+
+jobs:
+  # Tests against Qt/Python packages from PyPI
+  pip-qt:
+    env:
+      # Enforce selection of toolkit
+      ETS_TOOLKIT: qt
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        toolkit: ['pyside2']
+        python-version: [3.9]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Qt dependencies for Linux
+        uses: ./.github/actions/apt-get-qt-deps
+        if: startsWith(matrix.os, 'ubuntu')
+      - name: Update pip, setuptools and wheel
+        run: python -m pip install --upgrade pip setuptools wheel
+      - name: Install local packages
+        run: python -m pip install .[${{ matrix.toolkit }},editors,test]
+      - name: Run test suite
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: python -m unittest discover -v traitsui
+          working-directory: ${{ runner.temp }}
+
+  # Tests against wxPython from PyPI
+  # On OSX:
+  #   - wxPython complains about 'This program needs access to the screen'.
+  # On Ubuntu:
+  #   - Prebuilt wheels from wxpython.org are not compatible for our Python 3.8
+  #     and 3.9 setup. We can build from source and do something about
+  #     it taking a long time to build.
```

Comment:
The first sentence was actually only true when I tested with wxPython 4.1.0. wxPython 4.1.1 was released after this comment was written and the wheels for wxPython 4.1.1 are compatible with Python 3.8 and Python 3.9 here (I tested them). I will update this comment.